src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/onetimecode/OneTimeCodeService.kt

```diff
@@ -1,17 +1,42 @@
 package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.onetimecode
 
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.cjsm.CjsmService
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.config.ResourceNotFoundException
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.security.JwtService
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.security.SmokeTestConfig
 
 @Service
 class OneTimeCodeService(
   private val oneTimeCodeGenerator: OneTimeCodeGenerator,
   private val oneTimeCodeRepository: OneTimeCodeRepository,
   private val oneTimeCodeEmailSender: OneTimeCodeEmailSender,
+  private val jwtService: JwtService,
+  private val cjsmService: CjsmService,
+  private val smokeTestConfig: SmokeTestConfig,
 ) {
 
   fun createAndSendOneTimeCode(email: String, sessionId: String) {
     OneTimeCode(sessionId, oneTimeCodeGenerator.generateOneTimeCode(), email)
       .also { oneTimeCode -> oneTimeCodeRepository.save(oneTimeCode) }
       .also { oneTimeCode -> oneTimeCodeEmailSender.send(email, oneTimeCode.code) }
   }
+
+  fun verifyOneTimeCode(code: String, sessionId: String): String =
+    if (code == smokeTestConfig.lsjSecret) {
```

Comment:
This doesn't feel very secure, does this mean there is a magic version that disables some security features?

Reply:
Kind of, yes. The smoke test secret is a k8s secret. We have the secret defined in dev and preprod, but not prod.
For this to be a risk in prod (and yes, I accept that the mere fact we have that code is a risk!) we would need to have defined the secret in prod and a bad actor would need to know what that secret was.
Not ideal, but for our smoke tests we need to be able to 'authenticate' our smoke test user without it having a CJSM email address and without it being able to receive and read email